### PR TITLE
Apply jwt token to fetch

### PR DESCRIPTION
**Pull request by heartjigi**
Apply jwt token to fetch and
Refactoring server util -> api util
change some styles
@ethkim @nlpenguin @ace0625 @MechanicKim @SarahhhC @lyw0149 @ifndefdeadmau5

### CODE REVIEW

File: src/components/Activity/Connected.js [hunk 1]
```diff
@@ -33,6 +28,16 @@ class Connected extends Component {
     };
   }
 
+  onRequestCallback(result, error) {
+    if (result) {
+      this.nRequestSuccess(result);
+    }
+
+    if (error) {
```
Comment (SarahhhC) [inline src/components/Activity/Connected.js:33]: `else if`

File: src/components/Activity/NewRequests.js [hunk 2]
```diff
@@ -30,27 +25,27 @@ class NewRequests extends Component {
...
+        isEmpty: newRequests.length === 0,
+      });
+    }
 
-    this.setState({
-      dataSource: this.state.dataSource.cloneWithRows(newRequests),
-      loaded: true,
-      isEmpty: newRequests.length === 0,
-    });
+    if (error) {
```
Comment (MechanicKim) [inline src/components/Activity/NewRequests.js]: if -> else if

File: src/components/Activity/NewRequestsRow.js [hunk 3]
```diff
@@ -29,35 +28,35 @@ I really appriciate that if you give some advices of career This is the end`,
...
-    // If not, call 'getActivityList' to refresh parent's listview
-    if (result.pending) {
+  onGetActivityCallback(result, error) {
+    if (result) {
       this.props.onSelect(result);
-    } else {
-      ServerUtil.getActivityList();
+    }
+
+    if (error) {
```
Comment (MechanicKim) [inline src/components/Activity/NewRequestsRow.js:24]: if -> else if

File: src/components/Activity/NewRequestsRow.js [hunk 4]
```diff
@@ -29,35 +28,35 @@ I really appriciate that if you give some advices of career This is the end`,
...
 
-  onRequestFail(error) {
-    if (error.code !== ErrorMeta.ERR_NONE) {
-      alert(error.msg);
+  onRequestCallback(result, error) {
+    if (result) {
+      MatchUtil.getActivityList(this.onGetActivityCallback.bind(this));
+    }
+
+    if (error) {
```
Comment (MechanicKim) [inline src/components/Activity/NewRequestsRow.js:32]: if -> else if

File: src/components/Activity/NewRequestsRow.js [hunk 5]
```diff
@@ -86,16 +85,17 @@ I really appriciate that if you give some advices of career This is the end`,
           <View style={[styles.borderContainer]}>
             <View>
               <Text style={[styles.message]}
-                    ellipsizeMode={'tail'}
-                    numberOfLines={this.state.expanded ? 0 : 2}>
+                ellipsizeMode={'tail'}
```
Comment (MechanicKim) [inline src/components/Activity/NewRequestsRow.js]: Maybe changing '{'tail'}' into '"tail"' is possible!

File: src/components/Chat/ChatPage.js [hunk 6]
```diff
@@ -172,15 +169,15 @@ class ChatPage extends Component {
...
+    if (result) {
+      this.setState({
+        opponentInfo: result,
+      });
+    }
 
-  onServerFail(error) {
-    if (error.code !== ErrorMeta.ERR_NONE) {
-      alert(error.msg);
+    if (error) {
```
Comment (MechanicKim) [inline src/components/Chat/ChatPage.js]: if -> else if

File: src/components/Activity/NewRequestsRow.js (hunk 4 above)
Comment (ace0625) [inline src/components/Activity/NewRequestsRow.js:32]: else if

File: src/components/Chat/ChatPage.js (hunk 6 above)
Comment (ace0625) [inline src/components/Chat/ChatPage.js]: else if

File: src/components/Login/FindPassStep1.js [hunk 7]
```diff
@@ -49,19 +44,21 @@ class FindPassStep1 extends Component {
...
+      });
+    }
 
-  onError(error) {
-    if (error.code != ErrorMeta.ERR_NONE) {
-      Alert.alert(
-        'Forgot password',
-        'Please check email address that you inputted.',
-      );
+    if (error) {
```
Comment (ace0625) [inline src/components/Login/FindPassStep1.js:48]: else if

File: src/components/Login/FindPassStep3.js [hunk 8]
```diff
@@ -60,31 +56,32 @@ class FindPassStep3 extends Component {
...
+      );
+      Actions.login();
+    }
 
-  onServerFail(error) {
-    Alert.alert(
-      'Forgot password',
-      error.msg,
-    );
+    if (error) {
```
Comment (ace0625) [inline src/components/Login/FindPassStep3.js]: else if

File: src/components/Login/Login.js [hunk 9]
```diff
@@ -120,93 +126,69 @@ class Login extends Component {
...
-    ServerUtil.initCallback(onLocalLoginSuccess, onLocalLoginFail);
-    ServerUtil.signIn(this.state.email, this.state.password);
+  onLoginCallback(result, error) {
+    if (result) {
+      AsyncStorage.setItem('token', result.access_token,
+      () => UserUtil.getMyProfile(this.onGetProfileCallback.bind(this)));
+    }
+
+    if (error) {
+      console.log(error);
```
Comment (ace0625) [inline src/components/Login/Login.js]: isn't this for debugging?

File: src/components/Login/Login.js [hunk 10]
```diff
@@ -120,93 +126,69 @@ class Login extends Component {
...
+      }
+    }
+
+    if (error) {
+      this.setState({ loaded: true });
+    }
+  }
+
+  onGetProfileCallback(profile, error) {
+    console.log(profile, error);
```
Comment (ace0625) [inline src/components/Login/Login.js]: delete

File: src/components/Login/SignUp.js [hunk 11]
```diff
@@ -24,28 +23,32 @@ class Login extends Component {
...
-    if (error.code !== ErrorMeta.ERR_NONE) {
-      alert(error.msg);
+    if (error) {
+      alert(JSON.stringify(error));
     }
+
   }
 
   render() {
     let onChangeEmail = (text) => { this.state.email = text; };
```
Comment (ace0625) [inline src/components/Login/SignUp.js:41]: I think this part should be with setSate

File: src/components/SignUp/CareerInfo.js [hunk 12]
```diff
@@ -140,8 +155,14 @@ class CareerInfo extends Component {
...
   }
 
-  onUploadCallback() {
-    this.onNextBtnPressed();
+  onUploadCallback(result, error) {
+    if (result) {
+      Actions.expertInfo({ me: this.props.me });
+    }
+
+    if (error) {
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js:159]: else if

File: src/components/SignUp/ExpertInfo.js [hunk 13]
```diff
@@ -66,9 +67,9 @@ class ExpertInfo extends Component {
     }
 
     let body = { help };
+    UserUtil.editHelp(this.onUploadCallback.bind(this), body);
     console.log(JSON.stringify(body));
```
Comment (ace0625) [inline src/components/SignUp/ExpertInfo.js]: delete

File: src/components/SignUp/Personality.js [hunk 14]
```diff
@@ -26,15 +26,13 @@ class Personality extends Component {
...
-      (error) => this.onRequestFail(error)
-    );
   }
 
-  onRequestSuccess(result) {
-    Actions.completed({ me: this.props.me });
+  onUploadCallback(result, error) {
+    if (result)
+      Actions.completed({ me: this.props.me });
+    if (error)
```
Comment (ace0625) [inline src/components/SignUp/Personality.js]: else if

File: src/components/userProfile/RequestPage.js [hunk 15]
```diff
@@ -63,17 +59,16 @@ class RequestPage extends Component {
...
-  }
-
-  onRequestFail(error) {
-    if (error.code != ErrorMeta.ERR_NONE) {
-      Alert.alert(error.msg);
-    }
+  onRequestCallback(result, error) {
+    if (result)
+      Actions.requestSent();
+    if (error)
```
Comment (ace0625) [inline src/components/userProfile/RequestPage.js]: else if

File: src/utils/MatchUtil.js [hunk 16]
```diff
@@ -0,0 +1,41 @@
+import {
+  AccessToken,
+  LoginManager,
+} from 'react-native-fbsdk';
+import apiUtil from './ApiUtil';
+
+class MatchUtil {
+  // Request to mentor
```
Comment (ace0625) [inline src/utils/MatchUtil.js:9]: add an empty space

File: src/utils/UserUtil.js [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+  LoginManager,
+} from 'react-native-fbsdk';
+import {
+  LoginMeta,
+  ErrorMeta,
+} from './ApiMeta';
+import apiUtil from './ApiUtil';
+
+class UserUtil {
+  // Sign In with Facebook.
```
Comment (ace0625) [inline src/utils/UserUtil.js:13]: add an empty line

File: src/utils/UserUtil.js [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+          } else {
+            AccessToken.getCurrentAccessToken()
+            .then((data) => {
+              let body = {};
+              body.platform_type = LoginMeta.LOGIN_TYPE_FB;
+              body.access_token = data.accessToken.toString();
+              apiUtil.requestPost(callback, 'API_LOGIN', body);
+            })
+            .catch((error) => {
+              //console.log(error);
```
Comment (ace0625) [inline src/utils/UserUtil.js]: delete

File: src/components/Main.js [hunk 19]
```diff
@@ -24,6 +24,7 @@ class Main extends Component {
     new SendBird({
       appId: APP_ID,
     });
+    console.log(props);
```
Comment (MechanicKim) [inline src/components/Main.js]: Please remove

File: src/components/SignUp/Personality.js [hunk 20]
```diff
@@ -26,15 +26,13 @@ class Personality extends Component {
...
+      alert(JSON.stringify(error));
```
Comment (MechanicKim) [inline src/components/SignUp/Personality.js:33]: We can use 'Alter' component.
Other files use 'alert'. So, there might be many works.
But I think it is better to use 'Alter' component.

File: src/components/userProfile/UserProfile.js [hunk 21]
```diff
@@ -38,10 +37,14 @@ class UserProfile extends Component {
       evalLoaded: false,
       connectPressed: false,
     };
+  }
 
-    ServerUtil.initCallback(
-      (result) => this.onRequestSuccess(result),
-      (error) => this.onRequestFail(error));
+  onReqestCallback(result, error) {
+    if (result)
```
Comment (MechanicKim) [inline src/components/userProfile/UserProfile.js]: It is good to use '{', '}'

File: src/utils/SurveyUtil.js [hunk 22]
```diff
@@ -13,18 +13,19 @@ class SurveyUtil {
...
     let url = this.HOST + this.API_GET_SURVEY + select;
-    console.log(url);
     let reqSet = {
       method: this.GET,
       headers: {
         'Content-Type': this.CONTENT_TYPE_URLENCODED,
       },
     };
+
+    if (jwt) reqSet.headers.access_token = jwt;
```
Comment (MechanicKim) [inline src/utils/SurveyUtil.js]: For legibility, Convention says that using '{', '}' is better.

File: src/components/Login/SignUp.js (hunk 11 above)
Comment (heartjigi) [reply src/components/Login/SignUp.js:41]: @MechanicKim

File: src/components/Login/Login.js [hunk 23]
```diff
@@ -7,26 +7,31 @@ import {
...
     this.state = {
       email: '',
       password: '',
       loaded: false,
+      tokenValid: false,
     };
+
+    AsyncStorage.getItem('token', (err, result) => {
+      console.log(err, result);
+      if (false)
```
Comment (SarahhhC) [inline src/components/Login/Login.js]: What is this?????

File: src/components/Login/Login.js [hunk 24]
```diff
@@ -7,26 +7,31 @@ import {
...
-
     this.state = {
       email: '',
       password: '',
       loaded: false,
+      tokenValid: false,
     };
+
+    AsyncStorage.getItem('token', (err, result) => {
+      console.log(err, result);
```
Comment (SarahhhC) [inline src/components/Login/Login.js]: delete console

File: src/components/Login/Login.js [hunk 25]
```diff
@@ -35,6 +40,7 @@ class Login extends Component {
     }
 
     let onChangeEmail = (text) => { this.state.email = text; };
+
```
Comment (SarahhhC) [inline src/components/Login/Login.js:42]: delete line

File: src/components/Login/SignUp.js [hunk 26]
```diff
@@ -24,28 +23,29 @@ class Login extends Component {
       password1: '',
```
Comment (SarahhhC) [inline src/components/Login/SignUp.js:23]: What is password1 and 2?? The name is strange...

File: src/components/Login/SignUp.js [hunk 27]
```diff
@@ -24,28 +23,29 @@ class Login extends Component {
...
+        Actions.login();
+      } else {
+        AsyncStorage.setItem('token', result.access_token, () => Actions.generalInfo());
+      }
     }
   }
 
   render() {
     let onChangeEmail = (text) => { this.state.email = text; };
+
```
Comment (SarahhhC) [inline src/components/Login/SignUp.js:42]: delete line

File: src/components/Login/SignUp.js [hunk 28]
```diff
@@ -24,28 +23,29 @@ class Login extends Component {
...
     let onChangePassword1 = (text) => { this.state.password1 = text; };
+
```
Comment (SarahhhC) [inline src/components/Login/SignUp.js:44]: delete line

File: src/components/SignUp/Personality.js [hunk 29]
```diff
@@ -46,25 +45,26 @@ class Personality extends Component {
...
 
-    for (let [index, value] of this.state.values.entries()) {
+    for (let i = 0; i < this.state.values.length; i++) {
+      const value = this.state.values[i];
 
       // If user set default value(0), add both of personalities
-      if (value === 0) {
-        for (let title of this.state.sliderTitle[index]) {
+      if (this.state.values[i] === 0) {
+        for (let title of this.state.sliderTitle[i]) {
```
Comment (SarahhhC) [inline src/components/SignUp/Personality.js:53]: How about about using const? (by. @HewonJeong )

File: src/utils/ApiMeta.js [hunk 30]
```diff
@@ -0,0 +1,47 @@
+export const LoginMeta = {
+  LOGIN_TYPE_FB: '1',
+  LOGIN_TYPE_LOCAL: '2',
```
Comment (SarahhhC) [inline src/utils/ApiMeta.js]: In back-end, we use local login type as '0'. FYI 😈

File: src/utils/ApiMeta.js [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+  API_EDIT_HELP: 'users/editHelp/',
+  API_EDIT_PERSONALITY: 'users/editPersonality/',
+  API_LOCAL_SIGNUP: 'users/localSignUp/',
+  API_LOCAL_SIGNIN: 'users/localSignIn/',
+  API_SECRET_CODE: 'users/secretCode/',
+  API_RESET_PASS: 'users/resetPassword/',
+  API_GET_REQUEST_SETTING: 'users/mentorMode/',
+  API_SET_REQUEST_SETTING: 'users/editMentorMode/',
+  API_SIGN_OUT: 'users/signOut/',
+  API_TOKEN: 'users/all/',
```
Comment (SarahhhC) [inline src/utils/ApiMeta.js]: Duplicated one. check line 15

File: src/utils/ApiMeta.js [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export const ErrorMeta = {
+  ERR_NONE: 'Undefined Error',
+  ERR_TOKEN_INVALID: 'Token has been expired. Try again',
+  ERR_SERVER_FAIL: 'Login again',
+  ERR_NO_LOGIN_TYPE: 'Server error. Try again',
+  ERR_FB_LOGIN: 'Login error from facebook',
+  ERR_FB_LOGIN_CANCELLED: 'Login cancelled',
```
Comment (SarahhhC) [inline src/utils/ApiMeta.js:44]: Can we cancel login??

File: src/utils/ApiUtil.js [hunk 33]
```diff
@@ -0,0 +1,95 @@
...
+            if (idx > 0) url += '&';
+            url += key + '=' + body[key];
+          }
+        );
+      } else if (httpMethod === 'POST') {
+        reqSet.body = JSON.stringify(body);
+      }
+    }
+
+    if (jwt) reqSet.headers.access_token = jwt;
```
Comment (SarahhhC) [inline src/utils/ApiUtil.js]: Divide it by two line.

File: src/utils/SurveyUtil.js (hunk 22 above)
Comment (SarahhhC) [inline src/utils/SurveyUtil.js]: Divide it by two line. And jwt is null here, do we need it??

File: src/utils/SurveyUtil.js [hunk 34]
```diff
@@ -13,18 +13,19 @@ class SurveyUtil {
...
+
     fetch(url, reqSet)
     .then((resp) => {
-            console.log(resp);
             if (resp.ok) {
```
Comment (SarahhhC) [inline src/utils/SurveyUtil.js]: indentation

File: src/utils/ApiMeta.js (hunk 32 above)
Comment (heartjigi) [reply src/utils/ApiMeta.js:44]: FB can.

File: src/components/Login/Login.js (hunk 25 above)
Comment (heartjigi) [reply src/components/Login/Login.js:42]: JSCS said This line is needed

File: src/components/Login/SignUp.js (hunk 27 above)
Comment (heartjigi) [reply src/components/Login/SignUp.js:42]: JSCS said This line is needed

File: src/components/Main.js (hunk 19 above)
Comment (heartjigi) [reply src/components/Main.js]: solved

**Comment by SarahhhC** [comment]
Thanks @heartjigi Nice work LGTM 👍

**Pull request merged**